meson.build

```diff
@@ -0,0 +1,85 @@
+project(
+  'scikit-learn',
+  'c', 'cpp', 'cython',
+  # TODO should the version be dynamically generated, scikit-image does it but
+  # not scipy for some reason
+  version: '1.5.0.dev0',
+  license: 'BSD-3',
+  meson_version: '>= 1.1.0',
+  default_options: [
+    'buildtype=debugoptimized',
+    'c_std=c99',
+    'cpp_std=c++14',
+  ],
+)
+
+cc = meson.get_compiler('c')
+cpp = meson.get_compiler('cpp')
+
+# Check compiler is recent enough (see "Toolchain Roadmap" for details)
+if cc.get_id() == 'gcc'
+  if not cc.version().version_compare('>=8.0')
+    error('scikit-image requires GCC >= 8.0')
+  endif
+elif cc.get_id() == 'msvc'
+  if not cc.version().version_compare('>=19.20')
+    error('scikit-image requires at least vc142 (default with Visual Studio 2019) ' + \
+          'when building with MSVC')
+  endif
+endif
+
+_global_c_args = cc.get_supported_arguments(
+  '-Wno-unused-function',
+)
+add_project_arguments(_global_c_args, language: ['c', 'cpp'])
+
+# We need -lm for all C code (assuming it uses math functions, which is safe to
+# assume for scikit-image). For C++ it isn't needed, because libstdc++/libc++ is
+# guaranteed to depend on it. For Fortran code, Meson already adds `-lm`.
+m_dep = cc.find_library('m', required : false)
+if m_dep.found()
+  add_project_link_arguments('-lm', language : 'c')
+endif
+
+# Hide symbols when building on Linux with GCC. For Python extension modules,
+# we only need `PyInit_*` to be public, anything else may cause problems. So we
+# use a linker script to avoid exporting those symbols (this is in addition to
+# Meson using `-fvisibility=hidden` for C and `-fvisibility-inlines-hidden` for
+# C++ code. See gh-15996 for details.
```

Comment:
Maybe yes, maybe no. SciPy had an issue because they included lots of Fortran code, external static libraries, etc. And they wanted to prune those symbols from the public symbol table, especially the fortran code since you cannot build Fortran with GCC visibility.

Reply:
Thanks a lot for your comments, they are really useful!

I ended up removing this part indeed, deciding this was probably not relevant for scikit-learn. For reference the scipy issue is https://github.com/scipy/scipy/issues/15996.